### PR TITLE
Fix path specs so logging works again

### LINKED ISSUE #2220: 1.8.0-b2 creates a visible "${sys:appHome}" folder in user's Home
**Describe the bug**
On launch, MapTool creates a **${sys:appHome}** folder in the user's Home directory, where it stores logs.

**To Reproduce**
1. Install MapTool v. 1.8.0-beta.2 on Linux using a .deb package.
2. Launch MapTool.
3. Open your Home directory and check the list of folders.

**Expected behavior**
Logs should be saved under /home/[user]/.maptool-rptools/logs

**MapTool Info**
- Version: 1.8.0-beta.2
- Install: New, .deb package

**Desktop:**
 - OS: Linux Mint Tricia
 - Version 19.3
 (base: Ubuntu 18.04 bionic)

**Additional context**
Debug info:
==== MapTool Information ====
MapTool Version: 1.8.0-beta.2
MapTool Home...: /home/tomek/.maptool-rptools
MapTool Install: /opt/maptool/lib/app
Max mem avail..: 476 MB
Max mem used...: 89 MB
Custom Property: -DMAPTOOL_DATADIR=.maptool-rptools

==== Java Information ====
Java Home......: /opt/maptool/lib/runtime
Java Vendor....: Azul Systems, Inc.
Java Version...: 14.0.2
Java Parameters: 
  -Djava.library.path=/opt/maptool/lib/app:/opt/maptool/bin
  -Djava.launcher.path=/opt/maptool/bin
  -Xss8M
  -Dsentry.environment=Production
  -Dfile.encoding=UTF-8
  -XX:+ShowCodeDetailsInExceptionMessages
  --add-opens=javafx.web/javafx.scene.web=ALL-UNNAMED
  --add-opens=javafx.web/com.sun.webkit=ALL-UNNAMED

==== OS Information ====
OS Name........: Linux
OS Version.....: 5.4.0-47-generic
OS Architecture: amd64
PATH...........: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
Number of Procs: 2

==== User Information ====
User Name: tomek
User Home: /home/tomek
User Dir.: /home/tomek

==== Network Interfaces ====
Display Name..: wlp2s0
Interface Name: wlp2s0
Address...: fe80:0:0:0:35be:ccda:a573:caf5%wlp2s0
Address...: 2a02:a311:803c:c680:d5ad:29c8:8925:156%wlp2s0
Address...: 2a02:a311:803c:c680:fb95:775d:7aaf:eaa4%wlp2s0
Address...: 192.168.0.199

Display Name..: lo
Interface Name: lo
Address...: 0:0:0:0:0:0:0:1%lo
Address...: 127.0.0.1

Host Address...: 127.0.1.1
Default Gateway: 192.168.0.1

==== Locale Information ====
Country.: Poland
Language: Polish
Locale..: Polish (Poland)
Variant.: 

==== Encoding Information ====
Default Locale:   pl_PL
Default Charset:  UTF-8
file.encoding:    UTF-8
sun.jnu.encoding: UTF-8
Default Encoding: UTF8

==== Display Information ====
Number of Displays: 1
Display 1: 1366x768(-1)

==== Internet Gateway Devices ====
	No IGDs Found!

Created logfile:
[maptool.log](https://github.com/RPTools/maptool/files/5209223/maptool.log)

### PR DESCRIPTION
**Pull request by Phergus**
Fixes #2220 .  Path spec was calling for a `appHome` system property which didn't exist and failed to include the `logs` folder.  This will need to be tested under Linux/MacOS.

---
This change is [Reviewable](https://reviewable.io/reviews/rptools/maptool/2320)

### CODE REVIEW

**Review by kayila — APPROVED** [review]
Seems legit to me.

**Review by kayila — COMMENTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @cwisniew and @JamzTheMan)

**Review by kayila — APPROVED** [review]

**Review by Azhrei — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320)* status: all files reviewed, 1 unresolved discussion (waiting on @cwisniew, @JamzTheMan, and @Phergus)

___
*[src/main/resources/log4j2.xml, line 12 at r1](https://reviewable.io/reviews/rptools/maptool/2320#-MN0W4FIC8atd-h3cKqG:-MN0W4FJ7KkZD_9n47yA:b-22fayr) ([raw file](https://github.com/rptools/maptool/blob/e5a2e05a643ee00579a716b81118d286d5653841/src/main/resources/log4j2.xml#L12)):*
> ```XML
>         <RollingFile
>                 name="LogFile"
>                 fileName="${sys:user.home}/.maptool-rptools/logs/maptool.log"
> ```

I'm not sure we should use "user.home".  If we do that, changing MAPTOOL_DATADIR via Preferences will not have any effect on log files.  Is there a way to insert an arbitrary string instead?  Then AppUtil.getDataDir() could be used instead (I think that's the right method).

File: src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -9,16 +9,16 @@
...
+                filePattern="${sys:user.home}/.maptool-rptools/logs/archive/maptool_%d{yyyy-MM-dd_HH-mm-ss}.log.zip"
                 ignoreExceptions="false">
             <PatternLayout pattern="%d{yyyy-MM-dd HH:mm:ss.SSS} %-5level %logger{36} - %msg%n"/>
             <Policies>
                 <OnStartupTriggeringPolicy/>
                 <SizeBasedTriggeringPolicy size="10 MB"/>
             </Policies>
             <DefaultRolloverStrategy max="10">
-                <Delete basePath="${sys:appHome}/archive">
+                <Delete basePath="${sys:user.home}/.maptool-rptools/logs/archive">
```
Comment (kayila) [inline src/main/resources/log4j2.xml:21]:
```suggestion
                fileName="${MAPTOOL_DATADIR}/logs/maptool.log"
                filePattern="${MAPTOOL_DATADIR}/logs/archive/maptool_%d{yyyy-MM-dd_HH-mm-ss}.log.zip"
                ignoreExceptions="false">
            <PatternLayout pattern="%d{yyyy-MM-dd HH:mm:ss.SSS} %-5level %logger{36} - %msg%n"/>
            <Policies>
                <OnStartupTriggeringPolicy/>
                <SizeBasedTriggeringPolicy size="10 MB"/>
            </Policies>
            <DefaultRolloverStrategy max="10">
                <Delete basePath="${MAPTOOL_DATADIR}/logs/archive">
```

**Review by Phergus — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @Azhrei, @cwisniew, @JamzTheMan, and @kayila)

___
*[src/main/resources/log4j2.xml, line 12 at r1](https://reviewable.io/reviews/rptools/maptool/2320#-MN0W4FIC8atd-h3cKqG:-MN0eC0pC5KoqEkvyFsV:b-896fix) ([raw file](https://github.com/rptools/maptool/blob/e5a2e05a643ee00579a716b81118d286d5653841/src/main/resources/log4j2.xml#L12)):*
<details><summary><i>Previously, Azhrei (Frank Edwards) wrote…</i></summary><blockquote>

I'm not sure we should use "user.home".  If we do that, changing MAPTOOL_DATADIR via Preferences will not have any effect on log files.  Is there a way to insert an arbitrary string instead?  Then AppUtil.getDataDir() could be used instead (I think that's the right method).
</blockquote></details>

Done.
___
*[src/main/resources/log4j2.xml, line 21 at r1](https://reviewable.io/reviews/rptools/maptool/2320#-MN-1OnEyoY08iZ7mslH-r1-21:-MN0eH2gCg-0InLEhlWT:b-896fix) ([raw file](https://github.com/rptools/maptool/blob/e5a2e05a643ee00579a716b81118d286d5653841/src/main/resources/log4j2.xml#L21)):*
<details><summary><i>Previously, kayila wrote…</i></summary><blockquote>

```suggestion
                fileName="${MAPTOOL_DATADIR}/logs/maptool.log"
                filePattern="${MAPTOOL_DATADIR}/logs/archive/maptool_%d{yyyy-MM-dd_HH-mm-ss}.log.zip"
                ignoreExceptions="false">
            <PatternLayout pattern="%d{yyyy-MM-dd HH:mm:ss.SSS} %-5level %logger{36} - %msg%n"/>
            <Policies>
                <OnStartupTriggeringPolicy/>
                <SizeBasedTriggeringPolicy size="10 MB"/>
            </Policies>
            <DefaultRolloverStrategy max="10">
                <Delete basePath="${MAPTOOL_DATADIR}/logs/archive">
```
</blockquote></details>

Done.

**Review by Phergus — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320#-:-MN0gNStCTxHV57jC6Av:bnfp4nl)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @Azhrei, @cwisniew, @JamzTheMan, and @kayila)

**Review by kayila — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320#-:-MN0hV4qCNXYOnbrgBnL:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @Azhrei, @cwisniew, and @JamzTheMan)

**Review by JamzTheMan — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320)* status: all files reviewed, 1 unresolved discussion (waiting on @Azhrei and @cwisniew)

**Review by JamzTheMan — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320)* status: all files reviewed, 1 unresolved discussion (waiting on @Azhrei and @cwisniew)

**Review by JamzTheMan — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320#-:-MN0if05-W0ThDucXX3n:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @Azhrei and @cwisniew)

**Review by Azhrei — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/2320)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @cwisniew)

**Pull request merged**